source/MRMesh/MRReadTIF.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+#ifndef MRMESH_NO_TIFF
+#include "MRMeshFwd.h"
+#include "MRExpected.h"
+#include "MRVector2.h"
+#include <filesystem>
+#include <string>
+
+namespace MR
+{
+
+struct TifParameters
+{
+    enum class SampleType
+    {
+        Uint,
+        Int,
+        Float,
+        Unknown
+    } sampleType{ SampleType::Unknown };
+
+    enum class ValueType 
+    {
+        Scalar,
+        RGB,
+        RGBA,
+        Unknown
+    } valueType{ ValueType::Unknown };
+
+    // size of internal data in file
+    int bytesPerSample = 0;
+
+    // true if tif file is tiled
+    bool tiled = false;
+    Vector2i tileSize;
+    int layers = 1;
+    // tile depth (if several layers)
+    int depth = 0;
+
+    // size of image if not layered, otherwise size of layer
+    Vector2i imageSize;
+
+    bool operator==( const TifParameters& ) const = default;
+};
+
+// returns true if given file is tif
+bool isTIFFile( const std::filesystem::path& path );
+
+// reads parameters of tif file
+Expected<TifParameters, std::string> readTifParameters( const std::filesystem::path& path );
+
+struct RawTifOutput
+{
+    // main output data, should be allocated
+    float* data{ nullptr };
```

Comment:
if the user allocates it, then it is better to pass the size of allocation not to write in not-allocated memory